src/sql/src/plan/error.rs

```diff
@@ -340,6 +341,16 @@ impl PlanError {
             Self::KafkaSourcePurification(e) => e.hint(),
             Self::TestScriptSourcePurification(e) => e.hint(),
             Self::LoadGeneratorSourcePurification(e) => e.hint(),
+            Self::UnknownColumn { table, similar, .. } => {
+                match &similar[..] {
+                    [] => None,
+                    [column] => Some(format!("The similarly named column {} does exist.", ColumnDisplay { table, column })),
+                    names => {
+                        let similar = names.into_iter().map(|column| ColumnDisplay { table, column }).join(", ");
+                        Some(format!("There are similarly named columns that do exist {similar}."))
+                    }
+                }
+            }
```

Comment:
I think it would be helpful if somewhere in the error message we mention that identifiers have to be quoted if they aren't lower case.

Reply:
Added it!